### PR TITLE
Simplify `defmt_macros`

### PR DESCRIPTION
**Pull request by Urhengulas**

### CODE REVIEW

File: macros/src/lib.rs [hunk 1]
```diff
@@ -1131,7 +1131,7 @@ impl Codegen {
             .iter()
             .filter_map(|frag| match frag {
                 Fragment::Parameter(param) => Some(param.clone()),
-                Fragment::Literal(_) => None,
+                _ => None,
```
Comment (jonas-schievink) [inline macros/src/lib.rs:1081]: This was exhaustive to cause a compiler error when new variants are added

**Review by jonas-schievink — APPROVED** [review]
Looks good, just one minor issue

File: macros/src/lib.rs (hunk 1 above)
Comment (Urhengulas) [reply macros/src/lib.rs:1081]: This makes sense and I didn't think about that. Iirc there was a second place where I did this.

Comment (Urhengulas) [reply macros/src/lib.rs]: Seems to be only this one. Should be good now!

**Pull request merged**